pkgs/development/python-modules/mapbox/default.nix

```diff
@@ -0,0 +1,31 @@
+{ lib
+, buildPythonPackage
+, boto3
+, cachecontrol
+, fetchPypi
+, iso3166
+, python-dateutil
+, requests
+, polyline
+, uritemplate
+}:
+
+buildPythonPackage rec {
+  pname = "mapbox";
+  version = "0.18.0";
+
+  src = fetchPypi {
```

Comment:
please check out from github, there's no tests

```
running build_ext

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
```

tests are important to ensure that your package isn't broken by dependency bumps

Reply:
@jonringer, I changed it to download from GitHub in my local copy, but I'm still getting the same result. Here's what I see on my terminal. It looks like the checkPhase is run after :

```
[nix-shell:~/upstream/nixpkgs]$ nix-build '.' -A pythonPackages.mapbox
these derivations will be built:
  /nix/store/m85vk83zysmw00rpidk3wirz850xc5nn-python2.7-mapbox-0.18.0.drv
these paths will be fetched (0.45 MiB download, 2.83 MiB unpacked):
  /nix/store/2yisszfj7idvib7klw864ps5gpvnzina-python2.7-attrs-19.3.0
  /nix/store/dx9isd8iaxgg65kh78fnb3binniq3sb6-python2.7-atomicwrites-1.3.0
  /nix/store/ivg0g1zzz4rgil96kknb9h44n4ii9k2f-python2.7-wcwidth-0.1.7
  /nix/store/p4y7hl44qg7zcvgcvzd91xjaziy2gyhi-python2.7-coveralls-python-1.8.2
  /nix/store/v710zq8sxhnkfilrnmbg7n8znc2wvdrk-python2.7-docopt-0.6.2
  /nix/store/ybmnhismw51glvakzf2z7a5ixna3kgvp-python2.7-pytest-4.6.6
copying path '/nix/store/dx9isd8iaxgg65kh78fnb3binniq3sb6-python2.7-atomicwrites-1.3.0' from 'https://cache.nixos.org'...
copying path '/nix/store/2yisszfj7idvib7klw864ps5gpvnzina-python2.7-attrs-19.3.0' from 'https://cache.nixos.org'...
copying path '/nix/store/v710zq8sxhnkfilrnmbg7n8znc2wvdrk-python2.7-docopt-0.6.2' from 'https://cache.nixos.org'...
copying path '/nix/store/ivg0g1zzz4rgil96kknb9h44n4ii9k2f-python2.7-wcwidth-0.1.7' from 'https://cache.nixos.org'...
copying path '/nix/store/p4y7hl44qg7zcvgcvzd91xjaziy2gyhi-python2.7-coveralls-python-1.8.2' from 'https://cache.nixos.org'...
copying path '/nix/store/ybmnhismw51glvakzf2z7a5ixna3kgvp-python2.7-pytest-4.6.6' from 'https://cache.nixos.org'...
building '/nix/store/m85vk83zysmw00rpidk3wirz850xc5nn-python2.7-mapbox-0.18.0.drv'...
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
unpacking sources
unpacking source archive /nix/store/blhipijxh37rhmkb8nx05am724sjjspn-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/tox.ini
patching sources
configuring
no configure script, doing nothing
building
Executing setuptoolsBuildPhase
running bdist_wheel
The [wheel] section is deprecated. Use [bdist_wheel] instead.
running build
running build_py
creating build
creating build/lib
creating build/lib/mapbox
copying mapbox/__init__.py -> build/lib/mapbox
copying mapbox/errors.py -> build/lib/mapbox
copying mapbox/compat.py -> build/lib/mapbox
copying mapbox/encoding.py -> build/lib/mapbox
copying mapbox/utils.py -> build/lib/mapbox
creating build/lib/mapbox/services
copying mapbox/services/directions.py -> build/lib/mapbox/services
copying mapbox/services/static.py -> build/lib/mapbox/services
copying mapbox/services/__init__.py -> build/lib/mapbox/services
copying mapbox/services/datasets.py -> build/lib/mapbox/services
copying mapbox/services/maps.py -> build/lib/mapbox/services
copying mapbox/services/mapmatching.py -> build/lib/mapbox/services
copying mapbox/services/matrix.py -> build/lib/mapbox/services
copying mapbox/services/static_style.py -> build/lib/mapbox/services
copying mapbox/services/uploads.py -> build/lib/mapbox/services
copying mapbox/services/analytics.py -> build/lib/mapbox/services
copying mapbox/services/surface.py -> build/lib/mapbox/services
copying mapbox/services/base.py -> build/lib/mapbox/services
copying mapbox/services/tilequery.py -> build/lib/mapbox/services
copying mapbox/services/geocoding.py -> build/lib/mapbox/services
running egg_info
creating mapbox.egg-info
writing requirements to mapbox.egg-info/requires.txt
writing mapbox.egg-info/PKG-INFO
writing top-level names to mapbox.egg-info/top_level.txt
writing dependency_links to mapbox.egg-info/dependency_links.txt
writing manifest file 'mapbox.egg-info/SOURCES.txt'
reading manifest file 'mapbox.egg-info/SOURCES.txt'
writing manifest file 'mapbox.egg-info/SOURCES.txt'
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/mapbox
copying build/lib/mapbox/__init__.py -> build/bdist.linux-x86_64/wheel/mapbox
copying build/lib/mapbox/errors.py -> build/bdist.linux-x86_64/wheel/mapbox
copying build/lib/mapbox/compat.py -> build/bdist.linux-x86_64/wheel/mapbox
copying build/lib/mapbox/encoding.py -> build/bdist.linux-x86_64/wheel/mapbox
creating build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/directions.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/static.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/__init__.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/datasets.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/maps.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/mapmatching.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/matrix.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/static_style.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/uploads.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/analytics.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/surface.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/base.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/tilequery.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/geocoding.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/utils.py -> build/bdist.linux-x86_64/wheel/mapbox
running install_egg_info
Copying mapbox.egg-info to build/bdist.linux-x86_64/wheel/mapbox-0.18.0-py2.7.egg-info
running install_scripts
adding license file "LICENSE" (matched pattern "LICEN[CS]E*")
adding license file "AUTHORS" (matched pattern "AUTHORS*")
creating build/bdist.linux-x86_64/wheel/mapbox-0.18.0.dist-info/WHEEL
creating 'dist/mapbox-0.18.0-py2.py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'mapbox/__init__.py'
adding 'mapbox/compat.py'
adding 'mapbox/encoding.py'
adding 'mapbox/errors.py'
adding 'mapbox/utils.py'
adding 'mapbox/services/__init__.py'
adding 'mapbox/services/analytics.py'
adding 'mapbox/services/base.py'
adding 'mapbox/services/datasets.py'
adding 'mapbox/services/directions.py'
adding 'mapbox/services/geocoding.py'
adding 'mapbox/services/mapmatching.py'
adding 'mapbox/services/maps.py'
adding 'mapbox/services/matrix.py'
adding 'mapbox/services/static.py'
adding 'mapbox/services/static_style.py'
adding 'mapbox/services/surface.py'
adding 'mapbox/services/tilequery.py'
adding 'mapbox/services/uploads.py'
adding 'mapbox-0.18.0.dist-info/AUTHORS'
adding 'mapbox-0.18.0.dist-info/LICENSE'
adding 'mapbox-0.18.0.dist-info/METADATA'
adding 'mapbox-0.18.0.dist-info/WHEEL'
adding 'mapbox-0.18.0.dist-info/top_level.txt'
adding 'mapbox-0.18.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Finished executing setuptoolsBuildPhase
installing
Executing pipInstallPhase
/build/source/dist /build/source
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Processing ./mapbox-0.18.0-py2.py3-none-any.whl
Requirement already satisfied: iso3166 in /nix/store/ln5avxd5wnbg5l5n7lp3vlb761nmljjr-python2.7-iso3166-0.8/lib/python2.7/site-packages (from mapbox==0.18.0) (0.8)
Requirement already satisfied: uritemplate>=2.0 in /nix/store/9zp5hz482m18v9mig9rspi18zd79040n-python2.7-uritemplate-3.0.0/lib/python2.7/site-packages (from mapbox==0.18.0) (3.0.0)
Requirement already satisfied: python-dateutil>=2.5.0 in /nix/store/9qnh7ryw8yy0smk851qjq8kamzf3wnc4-python2.7-python-dateutil-2.8.1/lib/python2.7/site-packages (from mapbox==0.18.0) (2.8.1)
Requirement already satisfied: polyline>=1.3.1 in /nix/store/qdrw8wfgmzbw5px5n3jry6qx6lvww61c-python2.7-polyline-1.4.0/lib/python2.7/site-packages (from mapbox==0.18.0) (1.4.0)
Requirement already satisfied: cachecontrol in /nix/store/xdw9wwwb8gni9xrpfzm5achffljjxhl1-python2.7-CacheControl-0.12.5/lib/python2.7/site-packages (from mapbox==0.18.0) (0.12.5)
Requirement already satisfied: requests in /nix/store/xx5fa1fzblmg7c5raf3vzxa8pfcjn79j-python2.7-requests-2.22.0/lib/python2.7/site-packages (from mapbox==0.18.0) (2.22.0)
Requirement already satisfied: boto3>=1.4 in /nix/store/2kv45kdgq637lplf184lddg734qkjivy-python2.7-boto3-1.10.1/lib/python2.7/site-packages (from mapbox==0.18.0) (1.10.1)
Requirement already satisfied: six>=1.5 in /nix/store/2a86qa10xylfkqs1yg1vsx9cqr77m1vs-python2.7-six-1.12.0/lib/python2.7/site-packages (from python-dateutil>=2.5.0->mapbox==0.18.0) (1.12.0)
Requirement already satisfied: msgpack in /nix/store/1zaxmra8qn2dvf6cpkbsjnkrw293afwv-python2.7-msgpack-0.6.2/lib/python2.7/site-packages (from cachecontrol->mapbox==0.18.0) (0.6.2)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /nix/store/c8rhdq4ng13vzlcsx6lqbzxnf8gqilpz-python2.7-chardet-3.0.4/lib/python2.7/site-packages (from requests->mapbox==0.18.0) (3.0.4)
Requirement already satisfied: idna<2.9,>=2.5 in /nix/store/20j79v5sy05j4nya55ir95nbbr77bz1n-python2.7-idna-2.8/lib/python2.7/site-packages (from requests->mapbox==0.18.0) (2.8)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /nix/store/6nwjfzmgmn7b9lsd36mv8hv72jnwkwrr-python2.7-urllib3-1.25.7/lib/python2.7/site-packages (from requests->mapbox==0.18.0) (1.25.7)
Requirement already satisfied: certifi>=2017.4.17 in /nix/store/15qhda5wdf7l95ajvaawxcs6hlp998xm-python2.7-certifi-2019.9.11/lib/python2.7/site-packages (from requests->mapbox==0.18.0) (2019.9.11)
Requirement already satisfied: botocore<1.14.0,>=1.13.1 in /nix/store/8fhyg9v5q02bzm6ip6f9i18rj20x6178-python2.7-botocore-1.13.2/lib/python2.7/site-packages (from boto3>=1.4->mapbox==0.18.0) (1.13.2)
Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /nix/store/qsynbqp8fwy5239d1m581ggq9bjs21cp-python2.7-jmespath-0.9.4/lib/python2.7/site-packages (from boto3>=1.4->mapbox==0.18.0) (0.9.4)
Requirement already satisfied: s3transfer<0.3.0,>=0.2.0 in /nix/store/f4kbx6jbfk6fyq9dbb0d81f82gkzkciz-python2.7-s3transfer-0.2.1/lib/python2.7/site-packages (from boto3>=1.4->mapbox==0.18.0) (0.2.1)
Requirement already satisfied: docutils<0.16,>=0.10 in /nix/store/cba7q1n28689p7fa2ap4zhjbkhbkhff7-python2.7-docutils-0.15.2/lib/python2.7/site-packages (from botocore<1.14.0,>=1.13.1->boto3>=1.4->mapbox==0.18.0) (0.15.2)
Requirement already satisfied: futures<4.0.0,>=2.2.0; python_version == "2.6" or python_version == "2.7" in /nix/store/n8p74d1r0a17611sbmg1acw0gqglp1jf-python2.7-futures-3.3.0/lib/python2.7/site-packages (from s3transfer<0.3.0,>=0.2.0->boto3>=1.4->mapbox==0.18.0) (3.3.0)
Installing collected packages: mapbox
Successfully installed mapbox-0.18.0
/build/source
Finished executing pipInstallPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/nl62rfmcwbxw9l65sb977jkwgigdx3qw-python2.7-mapbox-0.18.0
strip is /nix/store/a0kdlyvbj9j4l67a8kxjr494qx6g3q0v-binutils-2.31.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/nl62rfmcwbxw9l65sb977jkwgigdx3qw-python2.7-mapbox-0.18.0/lib
patching script interpreter paths in /nix/store/nl62rfmcwbxw9l65sb977jkwgigdx3qw-python2.7-mapbox-0.18.0
checking for references to /build/ in /nix/store/nl62rfmcwbxw9l65sb977jkwgigdx3qw-python2.7-mapbox-0.18.0...
running install tests
no Makefile or custom buildPhase, doing nothing
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
setuptoolsCheckPhase
Executing setuptoolsCheckPhase
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
writing requirements to mapbox.egg-info/requires.txt
writing mapbox.egg-info/PKG-INFO
writing top-level names to mapbox.egg-info/top_level.txt
writing dependency_links to mapbox.egg-info/dependency_links.txt
reading manifest file 'mapbox.egg-info/SOURCES.txt'
writing manifest file 'mapbox.egg-info/SOURCES.txt'
running build_ext

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
Finished executing setuptoolsCheckPhase
pytestcachePhase
/nix/store/nl62rfmcwbxw9l65sb977jkwgigdx3qw-python2.7-mapbox-0.18.0

[nix-shell:~/upstream/nixpkgs]$
```

Reply:
Is this relevant?

```
Using setuptoolsCheckPhase
unpacking sources
unpacking source archive /nix/store/blhipijxh37rhmkb8nx05am724sjjspn-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/tox.ini
patching sources
configuring
no configure script, doing nothing
```

Reply:
this is what I would write:

```
{ lib, buildPythonPackage, fetchFromGitHub
, boto3
, cachecontrol
, iso3166
, polyline
, pytestCheckHook
, python-dateutil
, requests
, responses
, uritemplate
}:

buildPythonPackage rec {
  pname = "mapbox";
  version = "0.18.0";

  src = fetchFromGitHub {
    owner = "mapbox";
    repo = "mapbox-sdk-py";
    rev = version;
    sha256 = "123wsa4j11ps5pkjgylbmw4gnzh2vi22swgmvy50w26glkszh075";
  };

  propagatedBuildInputs = [ boto3 cachecontrol iso3166 python-dateutil requests polyline uritemplate ];

  checkInputs = [ pytestCheckHook responses ];

  meta = with lib; {
    homepage = https://github.com/mapbox/mapbox-sdk-py;
    license = licenses.mit;
    description = "Mapbox SDK for Python";
    longDescription = "Low-level client API for Mapbox web services.";
    maintainers = with maintainers; [ ersin ];
  };
}
```

Reply:
produces:
```
$ nix-build ./nixpkgs/ -A python3Packages.mapbox
these derivations will be built:
  /nix/store/hdfi0vq7wqhiq3lh4ic0ym8sam7hwd0a-python3.7-mapbox-0.18.0.drv
these paths will be fetched (0.08 MiB download, 0.30 MiB unpacked):
  /nix/store/8ls3vh1bx8mkwspc7dh0s9y7y5809nnb-python3.7-responses-0.10.6
  /nix/store/s2hvax59zmpvcrk4d0h06fqxrp1bfhd9-python3.7-cookies-2.2.1
copying path '/nix/store/s2hvax59zmpvcrk4d0h06fqxrp1bfhd9-python3.7-cookies-2.2.1' from 'https://cache.nixos.org'...
copying path '/nix/store/8ls3vh1bx8mkwspc7dh0s9y7y5809nnb-python3.7-responses-0.10.6' from 'https://cache.nixos.org'...
building '/nix/store/hdfi0vq7wqhiq3lh4ic0ym8sam7hwd0a-python3.7-mapbox-0.18.0.drv'...
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
Sourcing pytest-check-hook
Using pytestCheckPhase
unpacking sources
unpacking source archive /nix/store/blhipijxh37rhmkb8nx05am724sjjspn-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/tox.ini
patching sources
configuring
no configure script, doing nothing
building
Executing setuptoolsBuildPhase
running bdist_wheel
The [wheel] section is deprecated. Use [bdist_wheel] instead.
running build
running build_py
creating build
creating build/lib
creating build/lib/mapbox
copying mapbox/compat.py -> build/lib/mapbox
copying mapbox/encoding.py -> build/lib/mapbox
copying mapbox/utils.py -> build/lib/mapbox
copying mapbox/errors.py -> build/lib/mapbox
copying mapbox/__init__.py -> build/lib/mapbox
creating build/lib/mapbox/services
copying mapbox/services/directions.py -> build/lib/mapbox/services
copying mapbox/services/maps.py -> build/lib/mapbox/services
copying mapbox/services/surface.py -> build/lib/mapbox/services
copying mapbox/services/mapmatching.py -> build/lib/mapbox/services
copying mapbox/services/analytics.py -> build/lib/mapbox/services
copying mapbox/services/base.py -> build/lib/mapbox/services
copying mapbox/services/matrix.py -> build/lib/mapbox/services
copying mapbox/services/uploads.py -> build/lib/mapbox/services
copying mapbox/services/__init__.py -> build/lib/mapbox/services
copying mapbox/services/geocoding.py -> build/lib/mapbox/services
copying mapbox/services/tilequery.py -> build/lib/mapbox/services
copying mapbox/services/static_style.py -> build/lib/mapbox/services
copying mapbox/services/datasets.py -> build/lib/mapbox/services
copying mapbox/services/static.py -> build/lib/mapbox/services
running egg_info
creating mapbox.egg-info
writing mapbox.egg-info/PKG-INFO
writing dependency_links to mapbox.egg-info/dependency_links.txt
writing requirements to mapbox.egg-info/requires.txt
writing top-level names to mapbox.egg-info/top_level.txt
writing manifest file 'mapbox.egg-info/SOURCES.txt'
reading manifest file 'mapbox.egg-info/SOURCES.txt'
writing manifest file 'mapbox.egg-info/SOURCES.txt'
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/mapbox
creating build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/directions.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/maps.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/surface.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/mapmatching.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/analytics.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/base.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/matrix.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/uploads.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/__init__.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/geocoding.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/tilequery.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/static_style.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/datasets.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/services/static.py -> build/bdist.linux-x86_64/wheel/mapbox/services
copying build/lib/mapbox/compat.py -> build/bdist.linux-x86_64/wheel/mapbox
copying build/lib/mapbox/encoding.py -> build/bdist.linux-x86_64/wheel/mapbox
copying build/lib/mapbox/utils.py -> build/bdist.linux-x86_64/wheel/mapbox
copying build/lib/mapbox/errors.py -> build/bdist.linux-x86_64/wheel/mapbox
copying build/lib/mapbox/__init__.py -> build/bdist.linux-x86_64/wheel/mapbox
running install_egg_info
Copying mapbox.egg-info to build/bdist.linux-x86_64/wheel/mapbox-0.18.0-py3.7.egg-info
running install_scripts
adding license file "LICENSE" (matched pattern "LICEN[CS]E*")
adding license file "AUTHORS" (matched pattern "AUTHORS*")
creating build/bdist.linux-x86_64/wheel/mapbox-0.18.0.dist-info/WHEEL
creating 'dist/mapbox-0.18.0-py2.py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'mapbox/__init__.py'
adding 'mapbox/compat.py'
adding 'mapbox/encoding.py'
adding 'mapbox/errors.py'
adding 'mapbox/utils.py'
adding 'mapbox/services/__init__.py'
adding 'mapbox/services/analytics.py'
adding 'mapbox/services/base.py'
adding 'mapbox/services/datasets.py'
adding 'mapbox/services/directions.py'
adding 'mapbox/services/geocoding.py'
adding 'mapbox/services/mapmatching.py'
adding 'mapbox/services/maps.py'
adding 'mapbox/services/matrix.py'
adding 'mapbox/services/static.py'
adding 'mapbox/services/static_style.py'
adding 'mapbox/services/surface.py'
adding 'mapbox/services/tilequery.py'
adding 'mapbox/services/uploads.py'
adding 'mapbox-0.18.0.dist-info/AUTHORS'
adding 'mapbox-0.18.0.dist-info/LICENSE'
adding 'mapbox-0.18.0.dist-info/METADATA'
adding 'mapbox-0.18.0.dist-info/WHEEL'
adding 'mapbox-0.18.0.dist-info/top_level.txt'
adding 'mapbox-0.18.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Finished executing setuptoolsBuildPhase
installing
Executing pipInstallPhase
/build/source/dist /build/source
Processing ./mapbox-0.18.0-py2.py3-none-any.whl
Requirement already satisfied: cachecontrol in /nix/store/6azsya43iiar7lygxqppg0jzgw92nx56-python3.7-CacheControl-0.12.5/lib/python3.7/site-packages (from mapbox==0.18.0) (0.12.5)
Requirement already satisfied: iso3166 in /nix/store/a4s4j2b8bh6xsh0awrxa873bzdxw5dph-python3.7-iso3166-0.8/lib/python3.7/site-packages (from mapbox==0.18.0) (0.8)
Requirement already satisfied: python-dateutil>=2.5.0 in /nix/store/3ci7lxib88x1wvsb71qxs7fh1kxfha3r-python3.7-python-dateutil-2.8.1/lib/python3.7/site-packages (from mapbox==0.18.0) (2.8.1)
Requirement already satisfied: polyline>=1.3.1 in /nix/store/c1dqwj42gpxwpzfcm7s76lbrz5y57vyh-python3.7-polyline-1.4.0/lib/python3.7/site-packages (from mapbox==0.18.0) (1.4.0)
Requirement already satisfied: boto3>=1.4 in /nix/store/qfh2y87an1a41a1g46ikmzpqkx89gzl2-python3.7-boto3-1.10.1/lib/python3.7/site-packages (from mapbox==0.18.0) (1.10.1)
Requirement already satisfied: requests in /nix/store/0qzdiqjqfk7y0lnkvf3nqh3sy8ylhp4s-python3.7-requests-2.22.0/lib/python3.7/site-packages (from mapbox==0.18.0) (2.22.0)
Requirement already satisfied: uritemplate>=2.0 in /nix/store/qvr4zd54d0gq19n0as2k7hdw9q9l3ca0-python3.7-uritemplate-3.0.0/lib/python3.7/site-packages (from mapbox==0.18.0) (3.0.0)
Requirement already satisfied: msgpack in /nix/store/ss6r6w9k9p93kvvlaagkzwb89q452q6a-python3.7-msgpack-0.6.2/lib/python3.7/site-packages (from cachecontrol->mapbox==0.18.0) (0.6.2)
Requirement already satisfied: six>=1.5 in /nix/store/g9dwrvkgqwy96wxn64gy526ricjx8giw-python3.7-six-1.12.0/lib/python3.7/site-packages (from python-dateutil>=2.5.0->mapbox==0.18.0) (1.12.0)
Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /nix/store/mf3wfr2f1rzscph286xxll68h5n8j6s3-python3.7-jmespath-0.9.4/lib/python3.7/site-packages (from boto3>=1.4->mapbox==0.18.0) (0.9.4)
Requirement already satisfied: s3transfer<0.3.0,>=0.2.0 in /nix/store/i77zqq4bl0fz03scf2w8jad8v3176qqb-python3.7-s3transfer-0.2.1/lib/python3.7/site-packages (from boto3>=1.4->mapbox==0.18.0) (0.2.1)
Requirement already satisfied: botocore<1.14.0,>=1.13.1 in /nix/store/x9vwkbaq8f81lb0nr9zv6w141in54gy5-python3.7-botocore-1.13.2/lib/python3.7/site-packages (from boto3>=1.4->mapbox==0.18.0) (1.13.2)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /nix/store/fhd53xka4z814jad89amdccp00lkvrv3-python3.7-chardet-3.0.4/lib/python3.7/site-packages (from requests->mapbox==0.18.0) (3.0.4)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /nix/store/bkpx268fp1m84n27nhjhq9kc2chd3la7-python3.7-urllib3-1.25.7/lib/python3.7/site-packages (from requests->mapbox==0.18.0) (1.25.7)
Requirement already satisfied: certifi>=2017.4.17 in /nix/store/3j4vxznb9ivl3sh62n0mgpzgqs3p19g7-python3.7-certifi-2019.9.11/lib/python3.7/site-packages (from requests->mapbox==0.18.0) (2019.9.11)
Requirement already satisfied: idna<2.9,>=2.5 in /nix/store/bc182nf2dpj98v8blsxp2kbm37r91fyr-python3.7-idna-2.8/lib/python3.7/site-packages (from requests->mapbox==0.18.0) (2.8)
Requirement already satisfied: docutils<0.16,>=0.10 in /nix/store/i022rkab5nq4rv1mxmvw6bfm8cciqmx4-python3.7-docutils-0.15.2/lib/python3.7/site-packages (from botocore<1.14.0,>=1.13.1->boto3>=1.4->mapbox==0.18.0) (0.15.2)
Installing collected packages: mapbox
Successfully installed mapbox-0.18.0
/build/source
Finished executing pipInstallPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/bfas7b1wx92m8dqc9316pab9vmgwpv89-python3.7-mapbox-0.18.0
strip is /nix/store/a0kdlyvbj9j4l67a8kxjr494qx6g3q0v-binutils-2.31.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/bfas7b1wx92m8dqc9316pab9vmgwpv89-python3.7-mapbox-0.18.0/lib
patching script interpreter paths in /nix/store/bfas7b1wx92m8dqc9316pab9vmgwpv89-python3.7-mapbox-0.18.0
checking for references to /build/ in /nix/store/bfas7b1wx92m8dqc9316pab9vmgwpv89-python3.7-mapbox-0.18.0...
running install tests
no Makefile or custom buildPhase, doing nothing
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
setuptoolsCheckPhase
Executing setuptoolsCheckPhase
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
writing mapbox.egg-info/PKG-INFO
writing dependency_links to mapbox.egg-info/dependency_links.txt
writing requirements to mapbox.egg-info/requires.txt
writing top-level names to mapbox.egg-info/top_level.txt
reading manifest file 'mapbox.egg-info/SOURCES.txt'
writing manifest file 'mapbox.egg-info/SOURCES.txt'
running build_ext

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
Finished executing setuptoolsCheckPhase
pytestCheckPhase
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.7.5, pytest-5.2.4, py-1.8.0, pluggy-0.13.0
rootdir: /build/source, inifile: pytest.ini
collected 262 items

tests/test_analytics.py .................                                [  6%]
tests/test_base.py ............                                          [ 11%]
tests/test_datasets.py .............                                     [ 16%]
tests/test_directions.py .........................                       [ 25%]
tests/test_encoding.py .........                                         [ 29%]
tests/test_geocoder.py ........................                          [ 38%]
tests/test_mapmatching.py .....                                          [ 40%]
tests/test_maps.py .................................                     [ 52%]
tests/test_matrix.py .......................                             [ 61%]
tests/test_staticmaps.py ..........                                      [ 65%]
tests/test_staticstyle.py ...................                            [ 72%]
tests/test_surface.py ....                                               [ 74%]
tests/test_tilequery.py .........................................        [ 89%]
tests/test_upload.py ...................                                 [ 96%]
tests/test_utils.py ........                                             [100%]

=============================== warnings summary ===============================
mapbox/utils.py:1
mapbox/utils.py:1
  /build/source/mapbox/utils.py:1: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
    from collections import Mapping, Sequence

/nix/store/x9vwkbaq8f81lb0nr9zv6w141in54gy5-python3.7-botocore-1.13.2/lib/python3.7/site-packages/botocore/awsrequest.py:624
  /nix/store/x9vwkbaq8f81lb0nr9zv6w141in54gy5-python3.7-botocore-1.13.2/lib/python3.7/site-packages/botocore/awsrequest.py:624: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
    class HeadersDict(collections.MutableMapping):

tests/test_geocoder.py:306
  /build/source/tests/test_geocoder.py:306: DeprecationWarning: invalid escape sequence \/
    re.compile('https:\/\/api\.mapbox\.com\/geocoding\/v5\/mapbox\.places\/.+\.json'),

tests/test_directions.py::test_direction_backwards_compat
  /build/source/mapbox/services/directions.py:200: MapboxDeprecationWarning: Use `geometries` instead of `geometry`
    errors.MapboxDeprecationWarning)

tests/test_directions.py::test_direction_backwards_compat
tests/test_directions.py::test_v4_profile_aliases
  /build/source/mapbox/services/directions.py:42: MapboxDeprecationWarning: Converting v4 profile to v5, use mapbox/cycling instead
    errors.MapboxDeprecationWarning)

tests/test_surface.py::test_surface
tests/test_surface.py::test_surface_geojson
tests/test_surface.py::test_surface_params
  /build/source/mapbox/services/surface.py:33: MapboxDeprecationWarning: The surface module will be removed in the next version. It has no replacement.
    "It has no replacement.", MapboxDeprecationWarning)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
======================= 262 passed, 10 warnings in 1.03s =======================
Finished executing pytestCheckPhase
pytestcachePhase
/nix/store/bfas7b1wx92m8dqc9316pab9vmgwpv89-python3.7-mapbox-0.18.0
```

Reply:
@jonringer I appreciate the help! But why doesn't `pytestCheckHook` automatically get run during the build phase? It's barely documented in the manual under section "15.17.2.2.7. Setup hooks": "The following are setup hooks specifically for Python packages. Most of these are used in buildPythonPackage...pytestCheckHook to run tests with pytest."

Reply:
because there's other test runners such as unittest or nosetests. So it doesn't always make sense to use pytest